hazelcast/src/main/java/com/hazelcast/core/IMap.java

```diff
@@ -1869,6 +2240,25 @@ boolean tryLock(K key, long time, TimeUnit timeunit, long leaseTime, TimeUnit le
      * <p>
      * The operation is not lock-aware. The {@code EntryProcessor} will process the entries no matter if the keys are
      * locked or not. For more details check <b>Entry Processing</b> section on {@link IMap} documentation.
+     *
+     * <p><b>Interactions with the data store</b>
+     * <p>
+     * For each entry not found in-memory {@link MapLoader#load(Object)}
+     * is invoked to load the value from the data store backing the map.
+     * Exceptions thrown by load fail the operation and are propagated
+     * to the caller.
+     * <p>
+     * If write-through persistence mode is configured, for each entry
+     * updated by the entryProcessor, before the updated value is stored
+     * in-memory, {@link MapStore#store(Object, Object)} is called to
+     * write the value into the data store. Exceptions thrown by store
+     * fail the operation and are propagated to the caller.
+     * <p>
+     * If write-through persistence mode is configured, for each entry
```

Comment:
write-behind stores can also throw `ReachedMaxSizeException`

Reply:
Good point, thanks.
Added an additional section to the writer methods.